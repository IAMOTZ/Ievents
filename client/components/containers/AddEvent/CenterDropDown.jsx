import React from 'react';
import PropTypes from 'prop-types';

const CenterDropDown = props => (
  <select
    id="centers"
    className="form-control ml-md-3"
    name="centerId"
    onChange={props.handleChange}
    defaultValue={props.defaultValue}
  >
    <option value="" name="centerId">choose a center</option>
    {
      props.centers.map(center => (
        <option
          value={center.id}
          key={center.id}
          name="centerId"
        >{center.name}
        </option>))
    }
  </select>
);

CenterDropDown.defaultProps = {
  defaultValue: 0,
  centers: [],
};

/* eslint-disable react/forbid-prop-types */
CenterDropDown.propTypes = {
  centers: PropTypes.array,
  defaultValue: PropTypes.number,
  handleChange: PropTypes.func.isRequired,
};

export default CenterDropDown;
