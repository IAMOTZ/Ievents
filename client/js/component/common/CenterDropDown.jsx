import React from 'react';

export default (props) => {
  const centerOptions = props.centers.map((center) => {
    if (center.id === props.selected) {
      return (<option value={center.id} key={center.id} name="centerId" selected>{center.name}</option>)
    } else {
      return (<option value={center.id} key={center.id} name="centerId">{center.name}</option>)
    }
  });
  return centerOptions;
}