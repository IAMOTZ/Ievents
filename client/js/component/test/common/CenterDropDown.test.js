import React from 'react';
import renderer from 'react-test-renderer';
import CenterDropDown from '../../common/CenterDropDown';

const props = {
  centers: [
    { id: 1, name: 'test-center1' },
    { id: 2, name: 'test-center2' },
    { id: 3, name: 'test-center3' },
  ],
};

describe('<CenterDropDown />', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<CenterDropDown {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
