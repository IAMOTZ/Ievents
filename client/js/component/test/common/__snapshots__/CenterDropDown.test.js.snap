// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CenterDropDown /> should render correctly 1`] = `
Array [
  <option
    name="centerId"
    value={1}
  >
    test-center1
  </option>,
  <option
    name="centerId"
    value={2}
  >
    test-center2
  </option>,
  <option
    name="centerId"
    value={3}
  >
    test-center3
  </option>,
]
`;
